router/router_test.go

```diff
@@ -108,6 +110,105 @@ func TestHeadersFromConfiguration(t *testing.T) {
 	assert.Equal(t, "Nice", res.Header.Get("New-Cool-Header"))
 }
 
+func TestHeadersFromCORSConfig(t *testing.T) {
+	mode.Set(mode.Prod)
+	db := testdb.NewDBWithDefaultUser(t)
+	defer db.Close()
+
+	config := config.Configuration{PassStrength: 5}
+	config.Server.Cors.AllowOrigins = []string{"---", "http://test.com"}
+
+	g, closable := Create(db.GormDatabase,
+		&model.VersionInfo{Version: "1.0.0", BuildDate: "2018-02-20-17:30:47", Commit: "asdasds"},
+		&config,
+	)
+	server := httptest.NewServer(g)
+
+	defer func() {
+		closable()
+		server.Close()
+	}()
+
+	req, err := http.NewRequest("GET", fmt.Sprintf("%s/%s", server.URL, "version"), nil)
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("Origin", "http://test.com")
+	assert.Nil(t, err)
+
+	res, err := client.Do(req)
+	assert.Nil(t, err)
+	assert.Equal(t, "http://test.com", res.Header.Get("Access-Control-Allow-Origin"))
+}
+
+func TestCORSHeaderRegex(t *testing.T) {
+	mode.Set(mode.Prod)
+	db := testdb.NewDBWithDefaultUser(t)
+	defer db.Close()
+
+	config := config.Configuration{PassStrength: 5}
+	config.Server.Cors.AllowOrigins = []string{"---", "^http://test\\d{3}.com$"}
+
+	g, closable := Create(db.GormDatabase,
+		&model.VersionInfo{Version: "1.0.0", BuildDate: "2018-02-20-17:30:47", Commit: "asdasds"},
+		&config,
+	)
+	server := httptest.NewServer(g)
+
+	defer func() {
+		closable()
+		server.Close()
+	}()
+
+	req, err := http.NewRequest("GET", fmt.Sprintf("%s/%s", server.URL, "version"), nil)
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("Origin", "http://test123.com")
+	assert.Nil(t, err)
+
+	res, err := client.Do(req)
+	assert.Nil(t, err)
+	assert.Equal(t, "http://test123.com", res.Header.Get("Access-Control-Allow-Origin"))
+}
+
+// We want headers in responseheaders to override the cors config, as it is more specific
```

Comment:
It is the opposite actually, if you set a valid origin 
```
config.Server.Cors.AllowOrigins = []string{"http://test123.com"}
```
the headers actually will be overriden by the cors config. I'm okay with this, but the comment must be fixed.
(only Access-Control-Allow-Origin will be overridden, the other will be overriden of http method options is used)